### PR TITLE
Fixes issue 381

### PR DESCRIPTION
**Pull request by regiskuckaertz**
This fixes the issue https://github.com/guardian/scribe/issues/381 where two characters are left off the selection after the indent command is executed.

This does not fix:
- the behaviour where multiple Ps are automatically converted into a single H2 (step 2 of the process outlined in the issue which is something to do with scribe-plugin-heading-command)
- FF only: the behaviour where the selection is lost and therefore the whole operation fails when `scribe.el` is focused (something to do with scribe-plugin-toolbar)

Not strictly related but I thought could come in handy: the changes in selection serve to spare the need of creating many `Selection` instances by calling the `update` method instead. In fact, I was wondering if `Selection` wouldn't be better as a singleton.

### CODE REVIEW

**Comment by rrees** [comment]
Could you add some unit tests to demonstrate the behaviour/fix?

**Comment by regiskuckaertz** [comment]
Sure, I'll work this out today

File: test/commands.spec.js [hunk 1]
```diff
@@ -590,6 +590,20 @@ describe('commands', function () {
...
+    givenContentOf('<p>|test<br>test<br>test|</p>', function () {
+      when('the command is executed', function () {
+        beforeEach(function () {
+          return executeCommand('indent');
+        });
+
+        it('should have no characters left unselected', function () {
+          var selection = new window.scribe.api.Selection();
+          var range = selection.range;
+          return expect(range.endOffset).to.equal(range.endContainer.length);
```
Comment (regiskuckaertz) [inline test/commands.spec.js]: I'm rather new to Chai. Is it how it should be done? I guess we'll see the test results.

**Comment by rrees** [comment]
The tests are failing, if you have the chance to look then I think I'd be happy to merge.

**Comment by regiskuckaertz** [comment]
This may be that I need to bring this PR up to date will the latest changes. I'll look into it when I get back next week. Thanks!

**Comment by regiskuckaertz** [comment]
oh, that is my own test failing. I'll see what's wrong this evening.

**Comment by regiskuckaertz** [comment]
Yeah!

**Comment by rrees** [comment]
Awesome, I'll look into releasing it soon

**Comment by rrees** [comment]
@regiskuckaertz second favour of the day, could you rebase with master so I can release this?

**Comment by regiskuckaertz** [comment]
Sure!

**Pull request closed without merging**